File: Go/primes.go
package main

import (
	"fmt"
	"math"
	"time"
)


func isPrime(n int, primes []int) bool {
	sq := int(math.Sqrt(float64(n)))
	for _, i := range primes {
		if i > sq {
			break
		}
		if n % i == 0 {
			return false
		}
	}
	return true
}

func main() {
	start := time.Now()
	var primes []int
	for i := 2; i < 1000000; i++ {
		if isPrime(i, primes) {
			primes = append(primes, i)
		}
	}
	fmt.Printf("%d primes were found\n", len(primes))
	fmt.Printf("Process finished in %dms\n", time.Now().Sub(start).Milliseconds())
	fmt.Println(primes)
}

File: Python/primes.py
import math
import time

primes = []


def is_prime(i):
    sq = math.sqrt(i)
    for j in primes:
        if j > sq:
            break
        if i % j == 0:
            return False
    return True


def main():
    start = time.time()
    for i in range(2, 1000000):
        if is_prime(i):
            primes.append(i)
    print(f"{len(primes)} primes were found")
    print(f"Finished in {int((time.time() - start) * 1000)} ms")


if __name__ == "__main__":
    main()
